lib/checks/canFindRoots.js

```diff
@@ -0,0 +1,39 @@
+const Node = require('../node');
+const resolvesToConstant = require('./resolvesToConstant.js');
+
+function canFindRoots(equation) {
+
+  // Equation should be in the form of factor * factor = 0, factor^power = 0, or identifier = 0
+  // e.g (x - 2)^2 = 0, x(x + 2)(x - 2) = 0
+  const left = equation.leftNode;
+  const right = equation.rightNode;
+
+  const zeroRightSide = Node.Type.isConstant(right)
+        && parseFloat(right.value) === 0;
+
+  const isMulOrPower = Node.Type.isOperator(left, '*') || Node.Type.isOperator(left, '^');
+
+  if (!(zeroRightSide && isMulOrPower)) {
+    return false;
+  }
+
+  // If the left side of the equation is multiplication, filter out all the factors
+  // that do not evaluate to constants because they do not have roots. If the
+  // resulting array is empty, there is no roots to be found. Do a similiar check
+  // for when the left side is a power node.
+  // e.g 2^7, 2, (33 + 89)
+
+  if (Node.Type.isOperator(left, '*')) {
+    const factors = left.args.filter(arg => !resolvesToConstant(arg));
+    if (factors.length === 0) {
```

Comment:
Instead you could just `return factors.length > 1`, and then you wouldn't need the return at the bottom of the function.